releasing/plugin-commands-deploy/src/createDeployFiles.ts

```diff
@@ -256,45 +257,47 @@ function convertResolvedDependencies (
 interface ResolveLinkOrFileResult {
   scheme: 'link:' | 'file:'
   resolvedPath: string
-  suffix?: `(${string})`
+  suffix?: string
 }
 
 function resolveLinkOrFile (spec: string, opts: Pick<ConvertOptions, 'lockfileDir' | 'projectRootDirRealPath'>): ResolveLinkOrFileResult | undefined {
-  // try parsing `spec` as `spec(peers)`
-  const hasPeers = /^(?<spec>[^()]+)(?<peers>\(.+\))$/.exec(spec)
-  if (hasPeers) {
-    const result = resolveLinkOrFile(hasPeers.groups!.spec, opts)
-    if (!result) return undefined
-    if (result.suffix) {
-      throw new Error(`Something goes wrong, suffix is not undefined: ${result.suffix}`)
-    }
-    result.suffix = hasPeers.groups!.peers as `(${string})`
-    return result
-  }
-
-  // try parsing `spec` as either @scope/name@pref or name@pref
-  const renamed = /^@(?<scope>[^@]+)\/(?<name>[^@]+)@(?<pref>.+)$/.exec(spec) ?? /^(?<name>[^@]+)@(?<pref>.+)$/.exec(spec)
-  if (renamed) return resolveLinkOrFile(renamed.groups!.pref, opts)
-
   const { lockfileDir, projectRootDirRealPath } = opts
 
   if (spec.startsWith('link:')) {
-    const targetPath = spec.slice('link:'.length)
+    const { id, peersSuffix } = dp.parseDepPath(spec.slice('link:'.length))
```

Comment:
This function is currently not used anywhere in the codebase. I don't think you need it especially after you change the function to only accept depPath. Using `dp.parse()` should be enough to get all the necessary data.

Reply:
Initially, I tried separating the parsing of `spec` from `depPath` in https://github.com/pnpm/pnpm/pull/8926/commits/bea7d302de2c061c74e0e01989580931a0182705, but it doesn't work because the `spec` are the values from `ResolvedDependencies` which may contain `peersSuffix`. The written tests passed, but deploying headless of pnpm@v8 failed.

So, in https://github.com/pnpm/pnpm/pull/8926/commits/c90191ae6337aa53ae82fd18237fc40eaaa24595, I fixed it by reconstructing `depPath` from `name` and `spec`.